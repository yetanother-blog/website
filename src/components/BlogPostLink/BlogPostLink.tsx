import React from 'react';
import styled, { useTheme } from 'styled-components';
import { Link as GatsbyLink } from 'gatsby';
import { Typography } from '../Typography/Typography';
import { Link } from '../Link/Link';

export type BlogPostType = 'repo' | 'guide';

export type Test = 'primary' | 'secondary';

interface BlogPostLinkProps {
  url: string;
  slug: string;
  title: React.ReactNode;
  excerpt: React.ReactNode;
  metaData: React.ReactNode;
  blogType: BlogPostType;
}

const StyledGatsbyLink = styled(GatsbyLink)`
  display: block;
  text-decoration: none;
  font-family: inherit;
  color: inherit;
  position: relative;
`;

export const BlogPostLink: React.FC<BlogPostLinkProps> = ({
  url,
  slug,
  title,
  excerpt,
  metaData,
  blogType,
}) => {
  const theme = useTheme();

  const mapTypeToVariant: { [index in BlogPostType]: Test } = {
    repo: 'primary',
    guide: 'secondary',
  };

  return (
    <StyledGatsbyLink to={url} key={slug}>
      <Typography variant="headline" mb={theme.space.xs}>
        {title}
      </Typography>
      {metaData}
      <Typography variant="text" mb={theme.space.m}>
        {excerpt}
      </Typography>
      <Link
        variant={mapTypeToVariant[blogType]}
        display="block"
        component="span"
        marginBottom={theme.space.xxl}
      >
        read more
      </Link>
    </StyledGatsbyLink>
  );
};
