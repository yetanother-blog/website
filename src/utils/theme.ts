import { DefaultTheme } from 'styled-components';

export const theme: DefaultTheme = {
  fontSizes: ['12px', '14px', '16px', '20px', '24px', '32px'],
  colors: {
    primary: '#00FF00',
    secondary: '#FB00FF',
    grey500: '#1C1C1C',
    grey400: '#ACADAC',
    grey300: '#F6F6F6',
  },
  breakpoints: ['576px', '768px', '992px', '1200px'],
  space: {
    zero: 0,
    xxs: 4,
    xs: 8,
    s: 12,
    m: 16,
    l: 20,
    xl: 40,
    xxl: 60,
    xxxl: 120,
  },
  radii: {
    s: 3,
  },
  width: {
    m: '620px',
    l: '940px',
    xl: '1200px',
  },
};
