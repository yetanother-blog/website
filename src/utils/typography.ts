import { CSSObject } from "styled-components";
import Typography from "typography";

const typography = new Typography({
  googleFonts: [
    {
      name: "Source Sans Pro",
      styles: ["400", "600"]
    },
    {
      name: "Source Code Pro",
      styles: ["600", "700"]
    }
  ],
  baseFontSize: "20px",
  baseLineHeight: 1.85,
  bodyFontFamily: ["Source Sans Pro", "sans-serif"],
  boldWeight: "600",
  overrideStyles: ({ adjustFontSizeTo }, options, styles) => ({
    a: {
      fontFamily: `"Source Code Pro", "sans-serif"`
    }
  })
});

// Hot reload typography in development.
if (process.env.NODE_ENV !== `production`) {
  typography.injectStyles();
}

export default typography;

export const rhythm = typography.rhythm;
export const scale = typography.scale;

type StyledScale = (values: number) => CSSObject;
export const styledScale = scale as StyledScale;
