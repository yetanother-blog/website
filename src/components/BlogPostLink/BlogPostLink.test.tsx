import React from 'react';
import { render } from '@testing-library/react';
import { BlogPostLink } from './BlogPostLink';
import { ThemeContext } from 'styled-components';
import { theme } from '../../utils/theme';

describe('BlogPostLink component', () => {
  it('should render a BlogPostLink w/ content', () => {
    const { container } = render(
      <ThemeContext.Provider value={theme}>
        <BlogPostLink
          url="/some-url"
          title="some title"
          slug="some slug"
          excerpt="Some more text..."
          blogType="guide"
          metaData={<div>Some Content </div>}
        />
      </ThemeContext.Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
