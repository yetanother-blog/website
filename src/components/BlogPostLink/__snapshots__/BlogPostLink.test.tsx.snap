// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogPostLink component should render a BlogPostLink w/ content 1`] = `
<div>
  <a
    class="sc-fzpans faoBU"
    href="/some-url"
  >
    <h2
      class="sc-AxirZ epqEum"
    >
      some title
    </h2>
    <div>
      Some Content 
    </div>
    <p
      class="sc-AxhCb gYeYR"
    >
      Some more text...
    </p>
    <span
      class="sc-AxmLO fUYNGj"
    >
      <span
        class="sc-AxheI jxxOzf"
      >
        read more
      </span>
    </span>
  </a>
</div>
`;
